### PR TITLE
Add documentation for kitchen_sink

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Add documentation for kitchen_sink

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/85592
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/kitchen_sink.markdown [hunk 1]
```diff
@@ -0,0 +1,90 @@
+---
+title: Demo
+description: Instructions on how to use the Platform demos with Home Assistant.
+ha_category:
+  - Other
+  - Update
+ha_release: 0.7
```
Comment (frenck) [inline source/_integrations/kitchen_sink.markdown]:
```suggestion
ha_release: 2023.2
```

File: source/_integrations/kitchen_sink.markdown [hunk 2]
```diff
@@ -0,0 +1,90 @@
+---
+title: Demo
+description: Instructions on how to use the Platform demos with Home Assistant.
```
Comment (frenck) [inline source/_integrations/kitchen_sink.markdown]: This ain't the demo anymore 😉

File: source/_integrations/kitchen_sink.markdown [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ha_category:
+  - Other
+  - Update
+ha_release: 0.7
+ha_quality_scale: internal
+ha_codeowners:
+  - '@home-assistant/core'
+ha_domain: demo
```
Comment (frenck) [inline source/_integrations/kitchen_sink.markdown]:
```suggestion
ha_domain: kitchen_sink
```

File: source/_integrations/kitchen_sink.markdown [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  - update
+  - vacuum
+  - water_heater
+  - weather
+ha_integration_type: integration
+---
+
+The `demo` platform allows you to use integrations which are providing a demo of their implementation. The demo entities are dummies but show you how the actual platform looks like. This way you can run own demonstration instance like the online [Home Assistant demo](/demo/) but combined with your own real/functional platforms.
+
+Available demo platforms:
```
Comment (frenck) [inline source/_integrations/kitchen_sink.markdown]: Demo?

File: source/_integrations/kitchen_sink.markdown [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+```yaml
+# Example configuration.yaml entry
+
+# To load all:
+demo:
+
+# To load for a specific integration:
+light:
+  - platform: demo
```
Comment (frenck) [inline source/_integrations/kitchen_sink.markdown]: Demo leftovers

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by emontnemery** [comment]
I completely screwed up when first opening this PR, it was just a copy of the `demo` documentation 🤦
Now it should be reviewable @frenck 👍

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Pull request merged**